backend/corpora/common/entities/dataset.py

```diff
@@ -185,7 +165,10 @@ def new_processing_status() -> dict:
         }
 
     def copy_csv_to_s3(self, csv_file: str) -> str:
-        s3_file = f"{get_cxg_bucket_path(self.deployment_directories[0])}-genesets.csv"
+        cxg_object_name = self.explorer_s3_uri.split("/")[3].split(".")[
```

Comment:
Seems like the `get_cxg_bucket_path` function below is a bit more robust than the above written parsing of the string. Why not use it?

Reply:
I recommend moving this into its own function or using `get_cxg_bucket_path`